crates/core/src/nats/nats_conn.rs

```diff
@@ -0,0 +1,23 @@
+use super::{ConnStreams, NatsClient, NatsError};
+
+#[derive(Debug, Clone)]
+pub struct NatsConn {
+    pub client: NatsClient,
+    pub streams: ConnStreams,
+}
+
+impl NatsConn {
+    pub async fn new(
+        conn_id: &str,
+        nats_url: &str,
+        nats_nkey: &str,
+    ) -> Result<Self, NatsError> {
+        let client = NatsClient::connect(nats_url, conn_id, nats_nkey).await?;
+        let streams = ConnStreams::new(&client).await?;
```

Comment:
We just need a list of streams, not another struct. What do you think?

```suggestion
        let streams = streams:get_all(&client);
```